components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -327,7 +327,15 @@ public Application updateApplication(ApplicationUpdateRequest updateRequest, Str
             }
             appDTO.setRequestObjectSignatureValidationEnabled(updateRequest.isRequireSignedRequestObject());
             appDTO.setRequirePushedAuthorizationRequests(updateRequest.isRequirePushedAuthorizationRequests());
-            appDTO.setTlsClientCertificateBoundAccessTokens(updateRequest.isTlsClientCertificateBoundAccessTokens());
+            if (updateRequest.isTlsClientCertificateBoundAccessTokens()) {
+                boolean isCertificateTokenBindEnabled = DCRDataHolder.getInstance().getTokenBinders().stream()
```

Comment:
lets change the variable name to something like `isCertificateTokenBinderAvailable`